apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -119,7 +120,11 @@ protected Boolean initialValue() {
     private final MetricRegistry metricRegistry;
     private Sampler sampler;
     boolean assertionsEnabled = false;
-    private static final WeakConcurrentMap<ClassLoader, String> serviceNameByClassLoader = new WeakConcurrentMap.WithInlinedExpunction<>();
+    private long lastSpanMaxWarningTimestamp;
+
+    private volatile TracerState tracerState = TracerState.STOPPED;
```

Comment:
Maybe add a small comment to explain why those variables are `volatile` and which object is used to synchronize access to them.